src/biogeophys/SoilHydrologyMod.F90

```diff
@@ -412,11 +412,11 @@ subroutine Infiltration(bounds, num_hydrologyc, filter_hydrologyc, num_urbanc, f
 
              !--  if all water evaporates, there will be no bottom drainage
              if (h2osfc(c) < 0.0) then
-                qflx_infl(c) = qflx_infl(c) + h2osfc(c)/dtime
+                qflx_infl(c) = qflx_infl(c) + frac_h2osfc(c)*h2osfc(c)/dtime
```

Comment:
**Comment from @martynpclark Aug 17, 2017**

Note here that `qflx_infl` already includes the flux over the non-ponded fraction, and here we need to add the flux over the ponded fraction. The fraction of ponded water was missing originally.